ale_linters/cpp/clangtidy.vim

```diff
@@ -9,19 +9,27 @@ call ale#Set('cpp_clangtidy_checks', ['*'])
 " This will disable compile_commands.json detection.
 call ale#Set('cpp_clangtidy_options', '')
 call ale#Set('c_build_dir', '')
+" TODO: This can be common to all clang checkers requiring .cpp files.
+call ale#Set('cpp_clangtidy_header_suffixes', ['h', 'hpp', 'hxx', 'tcc'])
+" Set this option to handle options in case of header files.
+call ale#Set('cpp_clangtidy_header_sourcefile', '')
+" Set this option to set manually what file suffix is consider a header.
+" Set this option to manually set some options for clang-tidy.
 
 function! ale_linters#cpp#clangtidy#GetExecutable(buffer) abort
     return ale#Var(a:buffer, 'cpp_clangtidy_executable')
 endfunction
 
-function! s:GetBuildDirectory(buffer) abort
-    " Don't include build directory for header files, as compile_commands.json
-    " files don't consider headers to be translation units, and provide no
-    " commands for compiling header files.
-    if expand('#' . a:buffer) =~# '\v\.(h|hpp)$'
-        return ''
+function! ale_linters#cpp#clangtidy#IsHeader(buffer) abort
+    let l:search = '\v\.(' . join(ale#Var(a:buffer,'cpp_clangtidy_header_suffixes'), '|') . ')$'
```

Comment:
This isn't safe, as the option values could contain special regular expression characters. I think you'll be better off using an endswith check here.

```vim
let l:filename = expand('#' . a:buffer)

for l:suffix in ale#Var(a:buffer, 'cpp_clangtidy_header_suffixes')
    let l:extension = '.' . l:suffix

    if l:filename[-len(l:extension):] ==# l:extension
        return 1
    endif
endfor

return 0
```